content/docs/reference/design-docs/train-sim-v3/driver-behavior-modules.en.md

```diff
@@ -3,36 +3,80 @@ title: "Driver behavior modules"
-Decision[decision]
+To perform this critical task, it needs access to additional context:
 
-BuilderInputs --> DriverBehaviorModule
-CallInputs --> DriverBehaviorModule --> Decision
+- the physical properties of the path, which are used to make coasting decisions, and to model natural forces.
+- a slowdown kind, dictating how time loss is distributed along the path
+- a slowdown coefficient, which is used to adjust how much time is lost
 
-```
+### API
 
-Driver behavior modules are responsible for making driving decisions. It's main responsibility, given the state of the train,
-is to provide a driving instruction (a braking or traction force). To perform this critical task, it needs access to additional context:
+One driver behavior module is instantiated per driving instruction.
+It takes at initialization:
+- a slowdown coefficient
+- the slowdown kind
+- the driving instruction
+- the path properties
 
-- the physical properties of the path, which are used to make coasting decisions
-- driving instructions
+It has two public methods:
+- `enact_decision(current_state: TrainState, t: float) -> (TrainState, float)`
 
-## Design decisions
+  Which returns what the next train state would be if there was only this one instruction to follow, and the time delta
+  to reach this state.
+
+- `truncate_integration_step(current_state: TrainState, potential_state: TrainState, t: float, dt: float) -> (TrainState, float)`
+
+  Which returns a state and time delta which respects the instruction, and is as close as possible to the potential state. 
+
+
+### Loop
+At a given train state, we know which driving instructions are enforced.
+
+For each enforced driving instruction, we query the corresponding driver behavior module.
+
+This gives a set of different train states. From this, we coalesce a single train state which respects all instructions.
+
+To do so, we:
+1. Find the states which are most constraining for "constraining properties" (speed and pantograph state).
+  - Most constraining state regarding speed is the one with the lowest acceleration (taking sign into account).
+  - Most constraining state regarding pantograph state is the one which sets the pantograph down the earliest.
+2. Interpolate the **constraining** states to the smallest `dt` they are associated with.
+3. Merge the **constraining** states into a single state: 
+  - for speed, we take the lowest acceleration
+  - for pantograph state, we take the earliest pantograph state
+  - other properties should be identical
+4. Submit this state for truncation to all driver behavior modules. 
+
+{{% pageinfo color="warning" %}}
+There is a heavy underlying assumption that "constraining properties" can be combined in a new state which
+is valid.
+This underlies the step 3. It is not yet clear if this assumption will always be valid in the future.
+{{% /pageinfo %}}
+
+Here is a schema summarizing the process:
+
+![Driver behavior modules](../reconciliation.svg)
```

Comment:
Missing file?

Reply:
Arf, works on my machine ™️ 
I don't know why but sometimes hugo doesn't want to show images

Reply:
I mean, the file isn't in the PR nor in the repo, probably it's not tracked by git?

Reply:
It is tracked in the main train sim v3 branch

Reply:
Ehhh, my bad!

Reply:
It does work on my machine as well, it should reference [this doc](https://github.com/OpenRailAssociation/osrd-website/blob/only-specs/content/docs/reference/design-docs/train-sim-v3/reconciliation.svg)